### ISSUE TITLE
Triplet feel does not support?

### ISSUE THREAD

**Issue by Paopan** [CoderLine/alphaTab#107]
Hi Daniel, I wrote a triplet fee tab in gp5 format.
After rendered, the triplet feeling in the original file disappeared.
Could you suggest any way to solve this?

Thank you in advance.

**Comment by Danielku15** [comment]
Hi. 
I was working on this topic today and added support for triplet feel rendering in the latest commits. The triplet feel information is not yet used in the audio generation, I have to analyze first how the different triplet feel types influence the audio playback and apply the durations accordingly. Here an example how the rendering looks like for all triplet feels supported by Guitar Pro and AlphaTab:
![image](https://cloud.githubusercontent.com/assets/674916/20031639/28fca0f6-a37b-11e6-8069-ff9316cd059c.png)

I will keep this issue open until also the audio playback is handling the triplet feel correct.

**Comment by Paopan** [comment]
Thanks for your prompt attention to support this very rare used musical rhythm.

**Comment by cheap-glitch** [comment]
Are you still planning on supporting triplet feel during playback? Would be awesome if it could also be toggled on/off dynamically.

**Comment by Danielku15** [comment]
@cheap-glitch  Yes it is still on my roadmap. I did already some investigations on how it _should_ behave on audio playback but it is rather tricky to align the playback properly. There seem to be quite some rules behind a simple configuration of triplet feels. 

The general rule seems to be that if the specified triplet feel combination is on the beat of the time signature the playback is changed. But already if the note is slightly off-set the triplet feel does not apply anymore. 

But I also could notice some special rules in Guitar Pro that some other note combinations also are considered for the triplet feel.

**Comment by cheap-glitch** [comment]
Nice to know this is still in the works!

If you can manage to at least implement support for the "classic" 8th triplet feel (two eight notes = a triplet made of one quarter and one eight note), it would already cover a lot of music styles (pretty much all of blues/rock/bluegrass songs).

> But I also could notice some special rules in Guitar Pro that some other note combinations also are considered for the triplet feel.

Not 100% sure about this, but I think you could safely ignore all triplets and sextuplets, since they "override" the default triplet construct that is applied everywhere.

**Comment by Danielku15** [comment]
I just finished in the last few mins the audio generation for the "happy flow" when the notes are properly aligned timing wise and exactly matching the triplet feel definition. 😉

**Comment by Danielku15** [comment]
Basic generation of triplet feel audio is now supported.
